🌱  Add Cluster and ClusterClass variable defaulting and validation (#5615)
kubernetes-sigs/cluster-api: docs/proposals/202105256-cluster-class-and-managed-topologies.md

  - all the reference must be in the same namespace of `metadata.Namespace`
  - `spec.workers.machineDeployments[i].class` field must be unique within a ClusterClass.
  - `ClusterClassVariable`:

>     - names must be unique, not empty and not start with `builtin`

Reviewer: @sbueringer when we consider complex types, this should be changed into "should note be `builtin`" (reserved variable name)
Same below

Reviewer 2: Yeah, but I think we can already go back now to `should not be builtin` or `should not equal builtin` given that https://github.com/kubernetes-sigs/cluster-api/pull/5615#discussion_r749990826 will solve the `builtin.<something` variable name problem

Reviewer 2: I think we're now at: "names must be: unique, not empty, not equal to `builtin` and should not contain dots"
(same below for updates)

  - all the reference must be in the same namespace of `metadata.Namespace`
  - `spec.workers.machineDeployments[i].class` field must be unique within a ClusterClass.
  - `ClusterClassVariable`:

>     - names must be unique, not empty and not start with `builtin` and should not contain dots

Reviewer 2: ```suggestion
    - names must be unique, not empty, not equal to builtin and should not contain dots
```
nit, same below in l.643

Sorry for the back and forth. This syncs it now with the implementation
